hudi-common/src/main/java/org/apache/hudi/common/util/FileIOUtils.java

```diff
@@ -226,16 +226,17 @@ public static String[] getConfiguredLocalDirs() {
 
   private static boolean isRunningInYarnContainer() {
     // These environment variables are set by YARN.
-    return System.getenv("CONTAINER_ID") != null;
+    return System.getenv("CONTAINER_ID") != null
+        && System.getenv("LOCAL_DIRS") != null;
```

Comment:
What's nuances, the logic seems not change.

Reply:
This change is to prevent hudi runs in k8s pod from following this logic. Because there is also  variable CONTAINER_ID in the k8s pod, it will go to this logic and throw an exception。